packages/account/src/Sections/Security/ApiToken/api-token.tsx

```diff
@@ -43,31 +45,33 @@ const ApiToken = observer(() => {
         }),
         {
             api_tokens: [],
+            is_loading: true,
+            is_success: false,
             error_message: '',
+            show_delete: false,
+            is_delete_loading: false,
+            is_delete_success: false,
         }
     );
 
+    const timeout_ref = React.useRef<NodeJS.Timeout | undefined>();
+
     React.useEffect(() => {
-        /**
-         * Fetch all API tokens
-         */
-        getApiToken();
-    }, [getApiToken]);
+        getApiTokens();
```

Comment:
I suggest never do an API call in useEffect without a cancel token or a check if the response is setting any state within the component as it may lead to memory leak problems 

Something like this:

`Can't perform a React state update on an unmounted component.
This is a no-op, but it indicates a memory leak in your application.
To fix, cancel all subscriptions and asynchronous tasks in the componentWillUnmount
method.`

Reply:
I see that in our code we are making api calls in multiple places maybe we can take up this initiative in future refactoring cards. As of now, `Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in the componentWillUnmount method` is not showing up as warning in console. Thanks